frameworks/kafka/src/main/java/com/mesosphere/sdk/kafka/api/KafkaZKClient.java

```diff
@@ -84,17 +90,38 @@ public JSONObject getTopic(String topicName) throws Exception {
     }
 
     public List<String> getBrokerEndpoints() {
-        List<String> endpoints = new ArrayList<>();
         try {
-            List<String> ids = zkClient.getChildren().forPath(IDS_PATH);
-            for (String id : ids) {
-                byte[] bytes = zkClient.getData().forPath(IDS_PATH + "/" + id);
-                JSONObject broker = new JSONObject(new String(bytes, StandardCharsets.UTF_8));
-                endpoints.add(broker.getString("host") + ":" + broker.getInt("port"));
-            }
+            return getBrokerEndpoints(PROTOCOL_NAME_PLAINTEXT);
         } catch (Exception ex) {
             log.error("Failed to retrieve broker endpoints with exception: ", ex);
         }
+        return Collections.emptyList();
+    }
+
+    public List<String> getBrokerTLSEndpoints() {
+        try {
+            return getBrokerEndpoints(PROTOCOL_NAME_TLS);
+        } catch (Exception ex) {
+            log.error("Failed to retrieve broker TLS endpoints with exception: ", ex);
+        }
+        return Collections.emptyList();
+    }
+
+    private List<String> getBrokerEndpoints(String protocolName) throws Exception {
+        List<String> endpoints = new ArrayList<>();
+
+        List<String> ids = zkClient.getChildren().forPath(IDS_PATH);
+        for (String id : ids) {
+            byte[] bytes = zkClient.getData().forPath(IDS_PATH + "/" + id);
+            JSONObject broker = new JSONObject(new String(bytes, StandardCharsets.UTF_8));
+            String mappedProtocolName = broker.getJSONObject("listener_security_protocol_map").getString(protocolName);
+            StreamSupport.stream(broker.getJSONArray("endpoints").spliterator(), false)
+                    .map(endpoint -> endpoint.toString())
+                    .filter(endpoint -> endpoint.startsWith(mappedProtocolName))
+                    .map(endpoint -> endpoint.substring((mappedProtocolName + "://").length()))
```

Comment:
Not in std library - https://stackoverflow.com/questions/3100295/java-equivalent-of-pythons-string-lstrip